yew/src/virtual_dom/vtag.rs

```diff
@@ -328,14 +430,43 @@ impl VTag {
         }
     }
 
-    fn apply_diffs(&mut self, ancestor: &Option<Box<Self>>) {
+    fn apply_diffs(&mut self, ancestor: &mut Option<Box<Self>>) {
         let element = self.reference.as_ref().expect("element expected");
 
-        // Update parameters
-        let changes = self.diff_attributes(ancestor);
-
         // apply attribute patches including an optional "class"-attribute patch
-        for change in changes {
+        match (
+            &mut self.attributes,
+            &mut ancestor.as_mut().map(|a| &mut a.attributes),
+        ) {
+            (Attributes::Vec(new), Some(Attributes::IndexMap(_))) => {
+                self.attributes = Attributes::new_indexmap(std::mem::take(new));
+            }
+            (Attributes::IndexMap(_), Some(Attributes::Vec(old))) => {
+                let old = std::mem::take(old);
+                match ancestor.as_mut() {
+                    Some(a) => {
+                        a.attributes = Attributes::new_indexmap(old);
+                    }
+                    None => unreachable!(),
+                };
+            }
+            _ => (),
+        };
+        for change in match (
+            &self.attributes,
+            &ancestor
+                .as_ref()
+                .map(|a| &a.attributes)
+                .unwrap_or(&Attributes::default()),
```

Comment:
If ancestor has no attributes but `self` has `IndexMap` attributes, this `default` will cause `unreachable!` to be hit

Reply:
Fixed in f492acfd116117d512e32fc94c8e30ea3a1e1d1e.